src/org/ods/orchestration/BuildStage.groovy

```diff
@@ -103,17 +112,140 @@ class BuildStage extends Stage {
                 logMessage += buildTailorMessage(failedReposCommaSeparated, LOG_CUSTOM_PART)
                 jiraMessage += buildTailorMessage(failedReposCommaSeparated, JIRA_CUSTOM_PART)
             }
+
+            def aquaCriticalVulnerabilityRepos = filterReposWithAquaCriticalVulnerability(repos)
+            if (aquaCriticalVulnerabilityRepos?.size() > 0) {
+                def securityVulnerabilityIssueKeys = createSecurityVulnerabilityIssues(aquaCriticalVulnerabilityRepos)
+                String aquaMessage = buildAquaSecurityVulnerabilityMessage(securityVulnerabilityIssueKeys)
+                logMessage += aquaMessage
+                jiraMessage += aquaMessage
+            }
+
             util.failBuild(logMessage)
-            // If we are not in Developer Preview or we have a Tailor failure raise an exception
-            if (!project.isWorkInProgress || tailorFailedRepos?.size() > 0) {
+            // If we are not in Developer Preview or we have a Tailor failure or a Aqua remotely exploitable
+            // vulnerability with solution found then raise an exception
+            if (!project.isWorkInProgress || tailorFailedRepos?.size() > 0
+                || aquaCriticalVulnerabilityRepos?.size() > 0) {
                 throw new IllegalStateException(jiraMessage)
             }
         }
-        def aquaCriticalVulnerabilityRepos = filterReposWithAquaCriticalVulnerability(repos)
-        if (aquaCriticalVulnerabilityRepos?.size() > 0) {
-            String aquaFiledMessage = "Aqua critical vulnerability with solution detected"
-            util.failBuild(aquaFiledMessage)
-            throw new IllegalStateException(aquaFiledMessage)
+    }
+
+    List createSecurityVulnerabilityIssues(List aquaCriticalVulnerabilityRepos) {
```

Comment:
A more suitable place for all this Jira-related code that creates and transitions issues would be the JiraUseCase.

Reply:
Not sure that is the case as this also contain some message building (we already have message building in this class).

Reply:
Never mind, I moved all related functionality to JiraUseCase.

Reply:
Fine now